test/py/RunClientServer.py

```diff
@@ -277,6 +283,12 @@ def main():
 
     generated_dirs = []
     for gp_dir in options.genpydirs.split(','):
+        if gp_dir == 'type_hints' and (sys.version_info < (3,7)):
+            print('Skipping \'type_hints\' test since python 3.7 or later is required')
+            continue
+        if gp_dir == 'enum' and (sys.version_info < (3,7)):
```

Comment:
Look at https://issues.apache.org/jira/browse/THRIFT-5885 for perhaps the fix here